src/utils/entities/articles.ts

```diff
@@ -8,11 +8,15 @@ const flagged: ArticleFlag[] = [
 ]
 
 export function getArticleUrl(article: NFTArticleInfos): string {
-  return article.slug && !isArticleFlagged(article)
+  return isValidSlug(article.slug) && !isArticleFlagged(article)
     ? `/article/${article.slug}`
     : `/article/id/${article.id}`
 }
 
 export function isArticleFlagged(article: NFTArticleInfos): boolean {
   return flagged.includes(article.flag)
-}
\ No newline at end of file
+}
+
+export function isValidSlug(slug: string): boolean {
```

Comment:
The rule can be extended to "is valid if it doesn't contain a slug".
There are a few other iterations of slugs not being valid when they contain a dot.